src/ZohoDatabaseSyncZoho.php

```diff
@@ -0,0 +1,225 @@
+     * @param AbstractZohoDao $zohoDao
+     * @param string $localTable
+     */
+    public function pushDataToZoho(AbstractZohoDao $zohoDao, $localTable, $update = false){
+
+            $fieldsMatching = $this->findMethodValues($zohoDao);
+            $tableName = ZohoDatabaseHelper::getTableName($zohoDao,$this->prefix);
+            $rowsDeleted = [];
+            $statement = $this->connection->createQueryBuilder();
+            $statement->select('zcrm.*');
+            if($update){
+                $statement->addSelect('l.field_name as updated_fieldname');
+            }
+            $statement->from($localTable, 'l')
+            ->join('l', $tableName, 'zcrm', 'zcrm.uid = l.uid')
+            ->where('l.table_name=:table_name')
+            ->setParameters([
+                'table_name' => $tableName
+            ]);
+            $results = $statement->execute();
+            /* @var $zohoBeans ZohoBeanInterface[] */
+            $zohoBeans = array();
+            while ($row = $results->fetch()) {
+                $beanClassName = $zohoDao->getBeanClassName();
+                /* @var $zohoBean ZohoBeanInterface */
+                if(isset($zohoBeans[$row['uid']])){
+                    $zohoBean = $zohoBeans[$row['uid']];
+                }else{
+                    $zohoBean = new $beanClassName();
+                }
+                if(!$update){
+                    foreach ($row as $columnName => $columnValue) {
+                        if (!in_array($columnName,['id','uid']) || isset($fieldsMatching[$columnName])) {
+                            $value = $this->formatValueToBeans($zohoDao->getModule(), $fieldsMatching, $columnName, $columnValue, null, $row['uid']);
+                           if($columnValue){
+                               $zohoBean->{$fieldsMatching[$columnName]['setter']}($value);
+                           }
+                        }
+                    }
+                    $zohoBeans[$row['uid']] =  $zohoBean;
+                    $rowsDeleted[] = $row['uid'];
+                } else{
+                    $columnName = $row['updated_fieldname'];
+                    $zohoBean->setZohoId($row['id']);
+                    if (!in_array($columnName,['id','uid']) || isset($fieldsMatching[$columnName])) {
+                        $value = $this->formatValueToBeans($zohoDao->getModule(), $fieldsMatching, $columnName, $row[$columnName], $row['id']);
+                        $zohoBean->{$fieldsMatching[$columnName]['setter']}($value);
+                        $zohoBeans[$row['uid']] = $zohoBean;
+                        $rowsDeleted[] = $row['uid'];
+                    }
+                }
+            }
+            $zohoDao->save($zohoBeans);
+            if(!$update){
+                foreach ($zohoBeans as $uid => $zohoBean) {
+                    $this->connection->update($tableName, [ 'id'=>$zohoBean->getZohoId(),'lastActivityTime'=> date("Y-m-d H:i:s") ], ['uid'=>$uid ]);
```

Comment:
Why do we need to update lastActivityTime? This seems dangerous to me as any modification in Zoho before this lastActivityTime will not be fetched from Zoho. We should really not update it at all.